docs/datasources/flow_datasource.adoc

```diff
@@ -0,0 +1,63 @@
+= Flow Datasource
+
+== General
+
+The Flow Datasource can be used to retrieved application and conversation related metrics derived from flow data stored by OpenNMS.
+
+The query editor supports the following functions:
+
+[options="header, autowidth"]
+|===
+| Category  | Name                    | Description
+| Combine   | `topN`                  | Limit the results to the Top N conversations/applications when sorted by total bytes.
+| Combine   | `includeOther`          | Combine and include other conversations/applications that fall outside of the Top N.
+| Filter    | `withExporterNode`      | Filter for flows related to the given exporter node.
+| Filter    | `withIfIndex`           | Filter for flows related to the given interface index.
+| Transform | `perSecond`             | Convert the values to a per/second rate.
+| Transform | `negativeEgress`        | Make the egress values negative.
+| Transform | `negativeIngress`       | Make the ingress values negative.
+| Transform | `asTableSummary`        | Generate a tabular summary instead of time series data.
+| Transform | `combineIngressEgress`  | Sum ingress and egress values together.
+|===
+
+== Template Queries
+
+=== exporterNodesWithFlows
+
+The `exporterNodesWithFlows()` query will return a list of nodes which have exported flow data to OpenNMS.
+The value of the template variable is the `Node ID` and the display text is the `Node Label`.
+
+The available arguments are as follows:
+[options="header, autowidth"]
+|===
+| Name                 | Default       | Description
+| exportedWithinMillis | `14400000`    | Range of time in which a exporter must have a sent a flow to be included.
+                                           Template queries are not tied to the time range currently applied to the dashboard, so this value is relative to the current time.
+                                           Defaults to 4 hours.
+|===
+
+For example, to query the exporters that have sent flows in the last 24 hours you can use the following expression:
+```
+exporterNodesWithFlows(86400000)
+```
+
+=== interfacesOnExporterNodeWithFlows
+
+The `interfacesOnExporterNodeWithFlows()` query will return a list of interface indexes which are associated with one or more flows on the given node.
+The value of the template variable is the `Interface Index` and the display text is the `Interface Name`.
+
+The available arguments are as follows:
+[options="header, autowidth"]
+|===
+| Name                 | Default     | Description
+| nodeCriteria         | `(none)`    | Node criteria (Node ID, or Node Foreign Souce and Node Foreign ID) of the node of which to  enumerate the interfaces.
```

Comment:
To us it is probably clear, what the values of `nodeCriteria` can be, but maybe an hint on the format is useful here?
E.g. `nodeCriteria: 5` vs `nodeCriteria: 'default:3242342342342'`

Reply:
I added some examples bellow.